hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKeySerializer.java

```diff
@@ -94,64 +122,78 @@ public int getSerializedSize() {
         return VARIABLE_DATA_SIZE;
     }
 
+    /** {@inheritDoc} */
     @Override
-    public int getTypicalSerializedSize() {
-        return TYPICAL_SIZE;
+    public int getSerializedSize(@NonNull final OnDiskKey<K> key) {
+        return codec.measureRecord(key.getKey());
     }
 
+    /** {@inheritDoc} */
     @Override
-    public long getCurrentDataVersion() {
-        return 1;
+    public int getTypicalSerializedSize() {
+        return TYPICAL_SIZE;
     }
 
+    // Key serialization
+
+    /** {@inheritDoc} */
     @Override
-    public int deserializeKeySize(@NonNull final ByteBuffer byteBuffer) {
+    public void serialize(@NonNull final OnDiskKey<K> key, final @NonNull WritableSequentialData out) {
+        // Future work: https://github.com/hashgraph/pbj/issues/73
         try {
-            return codec.measure(BufferedData.wrap(byteBuffer)) + 4;
-        } catch (IOException e) {
-            // Maybe log here?
-            return -1;
+            codec.write(key.getKey(), out);
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
         }
     }
 
     @Override
-    public OnDiskKey<K> deserialize(@NonNull final ByteBuffer byteBuffer, final long ignored) throws IOException {
-        final var buff = BufferedData.wrap(byteBuffer);
-        final var len = buff.readInt();
-        final var pos = buff.position();
-        final var oldLimit = buff.limit();
-        buff.limit(pos + len);
-        final var k = codec.parse(buff);
-        buff.limit(oldLimit);
-        Objects.requireNonNull(k);
-        return new OnDiskKey<>(md, k);
+    @Deprecated(forRemoval = true)
+    public int serialize(final OnDiskKey<K> key, final ByteBuffer buffer) throws IOException {
+//        final BufferedData out = BufferedData.wrap(buffer);
+//        codec.write(key.getKey(), out);
+        throw new UnsupportedOperationException();
```

Comment:
Instead of having this method implemented in all the classes, you can have a default implementation in the interface that throws `UnsupportedOperationException` for these `serialize` and `deserialize` version.

Reply:
It isn't supported in just a very few places, like this. For majority of serializers it has to be implemented properly to enable migration from JDB